packages/react/src/lib/AutoControlledComponent.tsx

```diff
@@ -169,14 +178,19 @@ export default class AutoControlledComponent<P = {}, S = {}> extends UIComponent
       return acc
     }, {})
 
-    this.state = { ...state, ...initialAutoControlledState, autoControlledProps }
+    this.state = {
+      ...state,
+      ...initialAutoControlledState,
+      autoControlledProps,
+      getAutoControlledStateFromProps,
+    }
   }
 
   static getDerivedStateFromProps(props, state) {
-    const { autoControlledProps } = state
+    const { autoControlledProps, getAutoControlledStateFromProps } = state
```

Comment:
The single downside that I can't access actual `getAutoControlledStateFromProps()` by referencing `this` because it's static. Has anyone better idea than keep it in the state?

Reply:
I got confused, where this method comes from? Why are we extracting it form the state?

Reply:
Let me check this part again: https://codesandbox.io/s/elastic-framework-w3kfk

I expect that this should work properly...

Reply:
React calls it without `this`, that's why it's not working:
https://github.com/facebook/react/blob/2b93d686e359c7afa299e2ec5cf63160a32a1155/packages/react-test-renderer/src/ReactShallowRenderer.js#L560